src/cpp/src/visual_language/pipeline.cpp

```diff
@@ -99,8 +99,14 @@ class ov::genai::VLMPipeline::VLMPipelineImpl {
         size_t request_id = 0;
         size_t block_size = 1; // not used
         bool enable_prefix_caching = false;
+
+        auto tokenized_chat_history = m_inputs_embedder->get_tokenized_chat_history();
         size_t history_size = m_language.get_tensor("attention_mask").get_shape().at(1);
         size_t inputs_embeds_size = inputs_embeds.get_shape().at(1);
+        // inputs_embeds contains whole history
+        if (inputs_embeds_size == tokenized_chat_history.size()) {
```

Comment:
I'm not sure that we can compare such things.
As far as I remember, tokenized prompt and final embedding can be of different length for Llava ? as each image token in tokenized representation can be substituted with multiple ones in final prompt embedding